frontend/src/metabase/entities/segments.js

```diff
@@ -7,17 +7,19 @@ import colors from "metabase/lib/colors";
 
 const Segments = createEntity({
   name: "segments",
+  nameOne: "segment",
   path: "/api/segment",
   schema: SegmentSchema,
 
   objectActions: {
-    setArchived: ({ id }, archived, opts) =>
+    setArchived: ({ id, revision_message }, archived, opts) =>
       Segments.actions.update(
         { id },
         {
           archived,
           // NOTE: this is still required by the endpoint even though we don't really use it
-          revision_message: archived ? "(Archive)" : "(Unarchive)",
+          revision_message:
+            revision_message || (archived ? "(Archive)" : "(Unarchive)"),
```

Comment:
I don't have much context to know if we care about this, but I realized that we were passing in a message from a modal and then overwriting it.

Reply:
`setArchived({ id }, archived, { revision_message: ... })` I like that! 

I just updated to that format here: https://github.com/metabase/metabase/pull/9957/commits/cff70ac950fa78a62e654fd5dbb54cdb1ca5a66a